### PR TITLE
Add_optim_profile

### PR DESCRIPTION
**Pull request by CGMossa**
As mentioned in #173, this adds a "hyper" optimised build profile, and a new argument to enable this.

### CODE REVIEW

**Comment by CGMossa** [comment]
Note that I've updated the `edition` used to 2021.

File: R/source.R [hunk 1]
```diff
@@ -437,10 +443,25 @@ generate_cargo.toml <- function(libname = "rextendr",
...
+      panic = "abort",
+      `codegen-units` = 1
+    )
   )
 }
 
+generate_cargo_config.toml <- function() {
+  to_toml(
+    build = list(
+      rustflags = array(c("-C", "target-cpu=native")),
```
Comment (Ilia-Kosenkov) [inline R/source.R:460]:
```suggestion
      rustflags = c("-C", "target-cpu=native"),
```

`array()` is only needed to distinguish scalar values from vectors of length `1` (by adding `dim` attribute).

**Comment by Ilia-Kosenkov** [comment]
Not a fan of `optim`, but rust profile names are strange (neither full nor abbreviated). So if there is no better (short) word, lets make it `optim`.

**Comment by CGMossa** [comment]
Yeah I don't like optin either. I'd like to wait for someone creative to present a better name for it.

**Review by Ilia-Kosenkov — APPROVED** [review]
LGTM

**Comment by Ilia-Kosenkov** [comment]
@CGMossa , can you resolve a conflict here?

**Review by Ilia-Kosenkov — APPROVED** [review]

**Comment by CGMossa** [comment]
Fixes #173

**Pull request merged**